account/recovery.go

```diff
@@ -523,3 +525,53 @@ func GenerateRecoveryKeys(ctx context.Context, accountTarget uint32,
 	}
 	return acctKeys, nil
 }
+
+// AdvanceAccountDerivationIndex asserts that the internal wallet has at least
+// the given minimum index for deriving account keys.
+func AdvanceAccountDerivationIndex(ctx context.Context, minimumIndex uint32,
+	wallet lndclient.WalletKitClient, chainParams *chaincfg.Params) error {
+
+	allAccounts, err := wallet.ListAccounts(
+		ctx, "", walletrpc.AddressType_UNKNOWN,
+	)
+	if err != nil {
+		return fmt.Errorf("error listing accounts: %v", err)
+	}
+
+	poolAccountsPath := fmt.Sprintf("m/%d'/%d'/%d'",
+		keychain.BIP0043Purpose, chainParams.HDCoinType,
+		poolscript.AccountKeyFamily)
+
+	var poolAccountsAccount *walletrpc.Account
+	for idx, acct := range allAccounts {
+		if acct.DerivationPath == poolAccountsPath {
+			poolAccountsAccount = allAccounts[idx]
+		}
+	}
+
+	// If we did find our account in the list and already have enough keys
+	// derived, we don't need to do anything here.
+	if poolAccountsAccount != nil &&
+		poolAccountsAccount.ExternalKeyCount > minimumIndex {
+
+		log.Debugf("Account %s already has %d external keys (want "+
+			"minimum index %d), not deriving any keys",
+			poolAccountsPath, poolAccountsAccount.ExternalKeyCount,
+			minimumIndex)
+		return nil
+	}
+
+	for {
+		key, err := wallet.DeriveNextKey(
+			ctx, int32(poolscript.AccountKeyFamily),
+		)
+		if err != nil {
+			return fmt.Errorf("error deriving next key: %v", err)
+		}
+
+		log.Debugf("Derived next account key with index %d", key.Index)
+		if key.Index >= minimumIndex {
```

Comment:
Q: Do we want to end up with the last key being `minimumIndex` or the next one? Before we were comparing with `poolAccountsAccount.ExternalKeyCount > minimumIndex`

Reply:
Above it's the key _count_ while here it's the derived _index_. So the count is always one higher than the index. Added a comment above to make it a bit more clear.